templates/jenkins-override.epp

```diff
@@ -1,5 +1,12 @@
-<%| Hash[String, Struct[{value => Any}]] $environment |-%>
+<%- | Hash[String, Struct[{value => Any}]] $environment,
+      Optional[Hash[String, Struct[{value => Any}]]] $dropin_config = undef
```

Comment:
Do you expect to place entries here? Or why do you have the struct as a value with just a single entry (`value`)?

I think for the env vars I kept it to the complex type (with `value`) out of backwards compatibility for the parameter but for a new parameter I'd consider making it simple.